libs/dmlf/examples/boston_housing_multiprocess_distributed_learning/main.cpp

```diff
@@ -0,0 +1,220 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "dmlf/distributed_learning/distributed_learning_client.hpp"
+#include "dmlf/networkers/muddle_learner_networker.hpp"
+#include "dmlf/simple_cycling_algorithm.hpp"
+#include "math/matrix_operations.hpp"
+#include "math/tensor.hpp"
+#include "ml/dataloaders/ReadCSV.hpp"
+#include "ml/dataloaders/tensor_dataloader.hpp"
+#include "ml/exceptions/exceptions.hpp"
+#include "ml/ops/loss_functions/cross_entropy_loss.hpp"
+#include "ml/optimisation/adam_optimiser.hpp"
+#include "ml/optimisation/sgd_optimiser.hpp"
+
+#include <algorithm>
+#include <chrono>
+#include <ctime>
+#include <iostream>
+#include <string>
+#include <thread>
+#include <utility>
+#include <vector>
+
+using namespace fetch::ml::ops;
+using namespace fetch::ml::layers;
+using namespace fetch::ml::distributed_learning;
+
+using DataType         = fetch::fixed_point::FixedPoint<32, 32>;
+using TensorType       = fetch::math::Tensor<DataType>;
+using VectorTensorType = std::vector<TensorType>;
+using SizeType         = fetch::math::SizeType;
+
```

Comment:
lets add a comment here explaining what is the overall goal of this example. And then lets do the same for the other 5 examples so that it's totally clear what is the difference for each one